### PR TITLE
build: enable noUnusedParameters check in aot.

### PR DESCRIPTION
**Pull request by DevVersion**
* Re-enables the `noUnusedParameters` check in the e2eapp and demo-app (AOT).

### CODE REVIEW

**Comment by crisbeto** [comment]
Should be able to enable this one as well https://github.com/angular/material2/blob/dfe8091d959f1bb54cc33a8d942cbe7ea4e3fe56/src/universal-app/tsconfig-build.json#L8

**Comment by DevVersion** [comment]
Done.

File: src/universal-app/tsconfig-prerender.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "stripInternal": false,
     "experimentalDecorators": true,
     // TODO(paul): Remove once Angular has been upgraded and supports noUnusedParameters in AOT.
```
Comment (josephperrott) [inline src/universal-app/tsconfig-prerender.json]: Should this comment be removed as it was in the other json files?

Comment (DevVersion) [reply src/universal-app/tsconfig-prerender.json]: Good point. Done

**Pull request merged**